sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -588,20 +588,29 @@ private[hive] class TestHiveQueryExecution(
 
   override lazy val analyzed: LogicalPlan = {
     val describedTables = logical match {
-      case CacheTableCommand(tbl, _, _, _) => tbl.table :: Nil
+      case CacheTableCommand(tbl, _, _, _) => (tbl.table, tbl.database) :: Nil
       case _ => Nil
     }
 
     // Make sure any test tables referenced are loaded.
     val referencedTables =
       describedTables ++
-        logical.collect { case UnresolvedRelation(ident) => ident.last }
+        logical.collect { case UnresolvedRelation(ident) => (ident.last, ident.init.lastOption) }
```

Comment:
and .. we `unapply` with `AsTableIdentifier` or call `IdentifierHelper.asTableIdentifier`.

Reply:
Seems it's not convenient and proper to use either of them. `AsTableIdentifier` requires `TestHiveQueryExecution` to extend `LookupCatalog` and implement `catalogManager`. And `IdentifierHelper` actually requires `IdentifierImpl` instead.

So I used `TableIdentifier` directly.